src/envs/behavior_options.py

```diff
@@ -0,0 +1,976 @@
+"""Hardcoded options for BehaviorEnv."""
+# pylint: disable=import-error
+
+from typing import Callable, Dict, List, Sequence, Tuple, Union, Optional
+import numpy as np
+from numpy.random._generator import Generator
+import scipy
+from predicators.src.structs import State, Array
+from predicators.src.utils import get_aabb_volume, get_closest_point_on_aabb
+
+try:
+    import pybullet as p
+    from igibson import object_states
+    from igibson.envs.behavior_env import BehaviorEnv
+    from igibson.external.pybullet_tools.utils import CIRCULAR_LIMITS
+    from igibson.objects.articulated_object import URDFObject
+    from igibson.utils.behavior_robot_planning_utils import (
+        plan_base_motion_br,
+        plan_hand_motion_br,
+    )
+    from igibson.utils import sampling_utils
+    from igibson.external.pybullet_tools.utils import (
+        get_aabb,
+        get_aabb_extent,
+    )
+except ModuleNotFoundError as e:
+    print(e)
+
+_ON_TOP_RAY_CASTING_SAMPLING_PARAMS = {
+    "max_angle_with_z_axis": 0.17,
+    "bimodal_stdev_fraction": 1e-6,
+    "bimodal_mean_fraction": 1.0,
+    "max_sampling_attempts": 50,
+    "aabb_offset": 0.01,
+}
+
+
+# NOTE: we need a type:ignore because we can't specify the type of env because
+# it would be a igibson.envs.behavior_env.BehaviorEnv, which is an optional
+# import and won't get imported on CI
+def get_body_ids(  # type: ignore
+    env,
+    include_self: bool = False,
+    grasping_with_right: bool = False,
+) -> List[int]:
+    """Function to return list of body_ids for all objects for collision
```

Comment:
why does `detect_collision` not call this function if this function is "for collision checking"?

Reply:
it's used for different collision checkings